### ISSUE TITLE
Unable to generate a valid mocks for `AppLifecycleState` enum 🤔 

### ISSUE THREAD

**Issue by kamilklyta** [dart-lang/mockito#655]
As you can see in the example below, the `AppLifecycleState` enum has been replaced by the `dynamic` keyword.

I managed to reproduce it only for this enum - which is strange 🤔. I also don't see any errors in the console.
___

### Example class:
```dart
import 'dart:ui';

class MyClass {
  void method(AppLifecycleState state) {}
}
```
### Generated mock:
```dart
// Mocks generated by Mockito 5.4.2 from annotations
// in app/test/example_test.dart.
// Do not manually edit this file.

// ignore_for_file: no_leading_underscores_for_library_prefixes
import 'package:mockito/mockito.dart' as _i1;

import 'example_test.dart' as _i2;

// ignore_for_file: type=lint
// ignore_for_file: avoid_redundant_argument_values
// ignore_for_file: avoid_setters_without_getters
// ignore_for_file: comment_references
// ignore_for_file: implementation_imports
// ignore_for_file: invalid_use_of_visible_for_testing_member
// ignore_for_file: prefer_const_constructors
// ignore_for_file: unnecessary_parenthesis
// ignore_for_file: camel_case_types
// ignore_for_file: subtype_of_sealed_class

/// A class which mocks [MyClass].
///
/// See the documentation for Mockito's code generation for more information.
class MockMyClass extends _i1.Mock implements _i2.MyClass {
  MockMyClass() {
    _i1.throwOnMissingStub(this);
  }

  @override
  void method(dynamic state) => super.noSuchMethod(
        Invocation.method(
          #method,
          [state],
        ),
        returnValueForMissingStub: null,
      );
}

```

**Comment by srawlins** [comment]
Curious. In order for the tooling to know about `dart:ui`, I suspect you have to run `flutter` instead of `dart` somewhere. I don't use `flutter` so I'm not sure the equivalent, but can you run `build_runner` via `flutter run build_runner build` or something like that?

**Comment by kamilklyta** [comment]
Hi @srawlins, same result for both the `dart` and `flutter` commands. I'll try to reproduce this on a new project and let you know if this (for some reason) applies only to the current project.

**Comment by yanok** [comment]
I cannot reproduce it. Tried creating a new app, added `MyClass` and a mock for it. Got result:

```dart
// Mocks generated by Mockito 5.4.2 from annotations
// in simple/test/myclass_test.dart.
// Do not manually edit this file.

// ignore_for_file: no_leading_underscores_for_library_prefixes
import 'dart:ui' as _i3;

import 'package:mockito/mockito.dart' as _i1;
import 'package:simple/myclass.dart' as _i2;

// ignore_for_file: type=lint
// ignore_for_file: avoid_redundant_argument_values
// ignore_for_file: avoid_setters_without_getters
// ignore_for_file: comment_references
// ignore_for_file: implementation_imports
// ignore_for_file: invalid_use_of_visible_for_testing_member
// ignore_for_file: prefer_const_constructors
// ignore_for_file: unnecessary_parenthesis
// ignore_for_file: camel_case_types
// ignore_for_file: subtype_of_sealed_class

/// A class which mocks [MyClass].
///
/// See the documentation for Mockito's code generation for more information.
class MockMyClass extends _i1.Mock implements _i2.MyClass {
  @override
  void method(_i3.AppLifecycleState? state) => super.noSuchMethod(
        Invocation.method(
          #method,
          [state],
        ),
        returnValueForMissingStub: null,
      );
}
```

So it did correct import `dart:io` and it did correctly reference `AppLifecycleState` enum. We need more info on how to reproduce it.

BTW, `flutter run build_runner build` doesn't work, but `dart run build_runner build` does. One just has to make sure it's `dart` executable that ships with the flutter SDK.

**Comment by yanok** [comment]
Same as #633?

**Comment by yanok** [comment]
I actually *can* reproduce it now if I run `dart run build_runner build` using `dart` binary from vanilla (non-Flutter) Dart SDK.

@kamilklyta Could you please check if `which dart` and `which flutter` point to files in the same directory? And if not, try running `[path to Flutter bin]/dart run build_runner build`. Thanks!

**Comment by kamilklyta** [comment]
Hi @yanok, I tested this for both the non-flutter dart sdk I used before and the dart binary from the flutter directory, and both give me the same incorrect result 🤔.

**Comment by kamilklyta** [comment]
I asked my colleagues to try the same and we all were unable to generate a valid result.

**Comment by yanok** [comment]
This seems to be very similar to https://github.com/dart-lang/mockito/issues/662, so I can only ask the same questions.

1. What OS/environment are you using? Isn't it Windows/WSL by any chance? Could you try it with another OS?
2. Versions used, maybe just upload `pubspec.lock` somewhere.
3. Could you please make a minimal repo I could clone and reproduce? Please try cloning in a fresh directory and reproducing yourself if possible.

**Comment by yanok** [comment]
Looking at the old issues found https://github.com/dart-lang/mockito/issues/579 that has the same problem. Turns out that's a bug in `package:build_resolvers` that was fixed in version 2.1.0.

@kamilklyta  Could you please check `build_resolvers` version you are using and try to upgrade if it's below 2.1.0?

**Comment by yanok** [comment]
Closing, as there is no feedback and I'm pretty confident it's a manifestation of `build_resolvers` issue.
